zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -424,8 +516,25 @@ protected void pRequest2Txn(int type, long zxid, Request request,
                 validatePath(path, request.sessionId);
                 nodeRecord = getRecordForPath(path);
                 checkACL(zks, request.cnxn, nodeRecord.acl, ZooDefs.Perms.WRITE, request.authInfo, path, null);
+
+                // first do a quota check if the path is in a quota subtree.
+                ZKDatabase zkDatabase = zks.getZKDatabase();
+                byte[] lastdata;
+                DataNode n = zkDatabase.getNode(path);
+                byte[] data = setDataRequest.getData();
+                if (n != null) {
+                    String lastPrefix = zkDatabase.getDataTree().getMaxPrefixWithQuota(path);
```

Comment:
we are adding computations even if it is not needed (for instance in my project I do not have quotas)
if there any way to not run this code if enforceQuota is not enabled ?

Reply:
> if there any way to not run this code if enforceQuota is not enabled ?

if enforceQuota is not enabled, this code will not been run. But if enabled, that's inevitable